core/src/main/scala/akka/persistence/cassandra/journal/TagWriters.scala

```diff
@@ -342,9 +377,63 @@ import akka.util.ByteString
 
   // protected for testing purposes
   protected def createTagWriter(tag: String): ActorRef = {
-    context.actorOf(
-      TagWriter.props(settings, tagWriterSession, tag).withDispatcher(context.props.dispatcher),
-      name = URLEncoder.encode(tag, ByteString.UTF_8))
+    context.watch(
+      context.actorOf(
+        TagWriter.props(settings, tagWriterSession, tag, self).withDispatcher(context.props.dispatcher),
+        name = URLEncoder.encode(tag, ByteString.UTF_8)))
   }
 
+  private def tellTagActor(tag: String, message: Any, snd: ActorRef): Unit = {
+    passivatingTagActors.get(tag) match {
+      case Some(buffer) =>
+        passivatingTagActors = passivatingTagActors.updated(tag, buffer :+ PassivateBufferEntry(message, Left(snd)))
+      case None =>
+        tagActor(tag).tell(message, snd)
+    }
+  }
+
+  private def askTagActor(tag: String, message: Any)(implicit timeout: Timeout): Future[Any] = {
+    passivatingTagActors.get(tag) match {
+      case Some(buffer) =>
+        val p = Promise[Any]()
+        passivatingTagActors = passivatingTagActors.updated(tag, buffer :+ PassivateBufferEntry(message, Right(p)))
+        p.future
+      case None =>
+        tagActor(tag).ask(message)
+    }
+  }
+
+  private def tagActorTerminated(ref: ActorRef): Unit = {
+    tagActors.collectFirst { case (tag, a) if a == ref => tag } match {
```

Comment:
we could use a watchWith with a message that includes the tag or get the tag from the name

Reply:
changed in https://github.com/akka/akka-persistence-cassandra/pull/542/commits/22ebb50cad5b6098de5a4442ddc3946660722b3d